### PR TITLE
[libc][stdio] Use proxy headers of stdio.h in src and test folders.

### PR DESCRIPTION
**Pull request by lntue**
https://github.com/llvm/llvm-project/issues/60481

### CODE REVIEW

File: libc/include/llvm-libc-macros/stdio-macros.h [hunk 1]
```diff
@@ -9,6 +9,30 @@
...
+
+#ifdef __cplusplus
+extern "C" FILE *stdin;
+extern "C" FILE *stdout;
+extern "C" FILE *stderr;
+#else
+extern FILE *stdin;
+extern FILE *stdout;
+extern FILE *stderr;
+#endif
```
Comment (michaelrj-google) [inline libc/include/llvm-libc-macros/stdio-macros.h:22]: should this be in `FILE.h`?

File: libc/test/src/__support/File/platform_file_test.cpp [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include "src/__support/File/file.h"
 #include "test/UnitTest/Test.h"
 
-#include <stdio.h> // For SEEK_* macros
+#include <hdr/stdio_macros.h> // For SEEK_* macros
```
Comment (michaelrj-google) [inline libc/test/src/__support/File/platform_file_test.cpp:12]:
```suggestion
#include "hdr/stdio_macros.h" // For SEEK_* macros
```

File: libc/test/src/stdio/sprintf_test.cpp [hunk 3]
```diff
@@ -17,17 +17,6 @@
 
 // TODO: Add a comment here explaining the printf format string.
 
-// #include <stdio.h>
-// namespace LIBC_NAMESPACE_DECL {
-// using ::sprintf;
-// }
-
-class LlvmLibcSPrintfTest : public LIBC_NAMESPACE::testing::Test {
```
Comment (michaelrj-google) [inline libc/test/src/stdio/sprintf_test.cpp:25]: I don't think this change is necessary, but if you'd like to do it it should be in a separate patch.

File: libc/test/src/__support/File/platform_file_test.cpp (hunk 2 above)
Comment (lntue) [reply libc/test/src/__support/File/platform_file_test.cpp:12]: Done

File: libc/test/src/stdio/sprintf_test.cpp (hunk 3 above)
Comment (lntue) [reply libc/test/src/stdio/sprintf_test.cpp:25]: It is needed to fix a race condition due to multiple tests below accessing to the same buffer, that somehow become very consistent with the new proxy headers when I test locally.

Comment (nickdesaulniers) [reply libc/test/src/stdio/sprintf_test.cpp:25]: Then it can land before this PR.

File: libc/hdr/stdio_overlay.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#define __NO_INLINE__ 1
+#define LIBC_SET_NO_INLINE
+#endif
+
+#include <stdio.h>
+
+#ifdef LIBC_OLD_FORTIFY_SOURCE
+#define _FORTIFY_SOURCE LIBC_OLD_FORTIFY_SOURCE
+#undef LIBC_OLD_FORTIFY_SOURCE
+#endif
```
[the remark this answers is not in the document]
Comment (lntue) [reply libc/hdr/stdio_overlay.h:40]: Because the main purpose of this PR is actually for the issue in the description:
https://github.com/llvm/llvm-project/issues/60481#issuecomment-2362165751

File: libc/test/src/stdio/ungetc_test.cpp [hunk 5]
```diff
@@ -6,6 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "hdr/stdio_macros.h"
```
[the remark this answers is not in the document]
Comment (lntue) [reply libc/test/src/stdio/ungetc_test.cpp:9]: Moved the target header to the top.

File: libc/test/src/stdio/sprintf_test.cpp (hunk 3 above)
Comment (lntue) [reply libc/test/src/stdio/sprintf_test.cpp]: Moved this file's change to https://github.com/llvm/llvm-project/pull/110624

**Review by nickdesaulniers — APPROVED** [review]
